auth_api_key/models/auth_api_key.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2018 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models, tools, _
+
+from odoo.addons.server_environment import serv_config
+
+from odoo.exceptions import ValidationError, AccessError
+
+
+class AuthApiKey(models.TransientModel):
+    _name = "auth.api.key"
+    _description = "API Key Retriever"
+
+    @api.model
+    @tools.ormcache("api_key")
+    def _retrieve_uid_from_api_key(self, api_key):
+        if not self.env.user.has_group("base.group_system"):
+            raise AccessError(_("User is not allowed"))
+
+        for section in serv_config.sections():
+            if section.startswith("api_key_") and serv_config.has_option(
+                    section, "key"
+            ):
+                if api_key != serv_config.get(section, "key"):
+                    continue
+
+                login_name = serv_config.get(section, "user")
+                uid = self.env["res.users"].search(
```

Comment:
@sbidoul Is it possible to modify to search also the active=False users? 

Note: The suggestion below is 83 char (too long), but suggestion on multi lines isn't possible
```suggestion
                uid = self.env["res.users"].with_context(active_test=False).search(
```

Reply:
I'm not sure. I kind of remember people (was it @guewen?) saying running code under an inactive user was a bad idea, but I'm don't remember why.

Reply:
@sbidoul I do not know on further versions but at least on V10.0 I have absolutely no troubles while running code with an inactive technical user....

Reply:
@lmignon any opinion on this?

Reply:
@sbidoul IMO we must keep the possibility to select an inactive user (it's under the responsibility of the integrator/administrator to select the most appropriate one active or inactive). We must keep in mind that (at least in odoo < 12) active users are paid users. That's why we use inactive users  in such use cases.

Reply:
> I'm not sure. I kind of remember people (was it @guewen?) saying running code under an inactive user was a bad idea, but I'm don't remember why.

The problem with inactive users is that they will be removed from the group that they are in as soon as you edit an ir.group. The m2m widget for users on the group will not include them - as they are inactive.

Reply:
> I do not know on further versions but at least on V10.0 I have absolutely no troubles while running code with an inactive technical user....

Neither on other versions, until you edit the other users on a group and wonder why suddenly it doesn't work anymore ;)